micro-rdk/src/common/data_manager.rs

```diff
@@ -234,54 +239,111 @@ pub struct DataSyncTask<StoreType> {
+}
+
 impl<StoreType> DataSyncTask<StoreType>
 where
     StoreType: DataStore,
 {
-    async fn read_messages_for_collector(
-        &self,
-        collector_key: &ResourceMethodKey,
-    ) -> Result<Vec<SensorData>, DataSyncError> {
-        let store_lock = self.store.lock().await;
-        let mut raw_messages: Vec<BytesMut> = vec![];
-        let mut reader = store_lock.get_reader(collector_key)?;
-        loop {
-            let next_message = reader.read_next_message()?;
-            if next_message.is_empty() {
-                break;
-            }
-            raw_messages.push(next_message);
-        }
-        let data: Result<Vec<SensorData>, prost::DecodeError> =
-            raw_messages.into_iter().map(SensorData::decode).collect();
-        Ok(data?)
-    }
-
     async fn run<'b>(&mut self, app_client: &'b AppClient) -> Result<(), AppClientError> {
         for collector_key in self.resource_method_keys.iter() {
-            let data = match self.read_messages_for_collector(collector_key).await {
-                Ok(data) => data,
+            let store_lock = self.store.lock().await;
+            let mut current_chunk: Vec<BytesMut> = vec![];
+            let mut chunk_size_counter: usize = 0;
+            let mut reader = match store_lock.get_reader(collector_key) {
+                Ok(reader) => reader,
                 Err(err) => {
-                    log::error!(
-                        "error decoding readings for collector key ({:?}): {:?}",
-                        collector_key,
-                        err
-                    );
+                    log::error!("error acquiring store reader: {:?}", err);
                     continue;
                 }
             };
-            let upload_request = DataCaptureUploadRequest {
-                metadata: Some(UploadMetadata {
-                    part_id: self.part_id.clone(),
-                    component_type: collector_key.component_type.clone(),
-                    r#type: DataType::TabularSensor.into(),
-                    component_name: collector_key.r_name.clone(),
-                    method_name: collector_key.method.to_string(),
-                    ..Default::default()
-                }),
-                sensor_contents: data,
-            };
-            app_client.upload_data(upload_request).await?;
+            loop {
+                let next_message = match reader.read_next_message() {
+                    Ok(msg) => msg,
+                    Err(err) => {
+                        log::error!("error reading from store: {:?}", err);
+                        break;
+                    }
+                };
+                if next_message.is_empty() {
+                    break;
+                }
+                // If the amount of data becomes too big for a single upload, we want to start a new
+                // collection of upload data
+                if (chunk_size_counter + next_message.len()) > MAX_SENSOR_CONTENTS_SIZE {
```

Comment:
Per discussion, this will drop a message on overflow. Additionally, there is no protection against encountering an entry that exceeds the max upload size, which would stall indefinitely if the above issue was fixed.